mod/01_core/format/html_format.rb

```diff
@@ -216,8 +265,8 @@ def unsupported_view view
     end
 
     def final_link href, opts={}
-      text = opts[:text] || href
-      %{<a class="#{opts[:class]}" href="#{href}">#{text}</a>}
+      text = (opts.delete(:text) || href).dup
+      content_tag :a, text, opts.merge(:href=>href)
```

Comment:
re "if you make the case that it is broken".  My take is that the case when we change this stuff we have to make the case that we're NOT breaking things with a lot of field testing.   If this dirty work falls to me, then it might be a while before we get this merged in.  I've just got to be cautious, because if I merge it in and it screws up urls all over the place, I'm going to have to do all the clean up.